### PR TITLE
Update HasOne-relations.md

### PR DESCRIPTION
**Pull request by juliandanciu**
This documentation chooses to explain the HasOne relationship with the Account and Supplier models as examples. Since we are documenting the HasOne relationship, it is my understanding that the DECLARING model is the Supplier model and the TARGET model is the Account model. The foreign key therefore would be on the declaring model as account.supplierId. I corrected the documentation to reflect that. Also, the type of the foreign key property (defined on the target model) should be the same as the type of the id key property on the declaring model (and not the target model as the docs state). I believe that the use of declaring and target has been flipped mistakenly. Of course, I'm just learning this stuff so I could be very wrong. If so, please disregard this proposed file change :)

### CODE REVIEW

**Review by dhmlau — APPROVED** [review]

**Comment by dhmlau** [comment]
@juliandanciu , thanks for your contribution.  Your PR looks good to me.  
Before we can land, could you please sign the CLA? https://cla.strongloop.com/agreements/strongloop/loopback.io

@b-admike @jannyHou , would like another pair of eyes just in case I got the declaring & target model mixed up. :)  Thanks!

**Comment by juliandanciu** [comment]
Hello Diana,

Thanks. I just signed the CLA.

Have a wonderful day.

**Review by jannyHou — APPROVED** [review]
🚢

**Comment by dhmlau** [comment]
Thanks @juliandanciu for your contribution.  Your PR has landed! 🎉

**Pull request merged**
